README.md
# Simple coverage badger

*Using shields.io*

## Usage
```
$ scb [options]

Options:
  --help                  Show help                                                                            [boolean]
  --version               Show version number                                                                  [boolean]
  --input, -i             Input file path                        [string] [default: "./coverage/cobertura-coverage.xml"]
  --outputFileName, -o    Output file name (without extension)                               [string] [default: "badge"]
  --outputDirectory, -d   Output file name (without extension)                                   [string] [default: "."]
  --format, -f            Output file format (extension)               [string] [choices: "svg", "png"] [default: "svg"]
  --badgeHosting, -h      Badge hosting server                              [string] [default: "https://img.shields.io"]
  --coverageScope, -c     Scope for badge                     [string] [choices: "branches", "lines"] [default: "lines"]
  --excelentTreshold, -e  Treshold for excelent coverage rating (brightgreen)                     [number] [default: 95]
  --goodTreshold, -g      Treshold for good coverage rating (green)                               [number] [default: 80]
  --mediumTreshold, -m    Treshold for medium coverage rating (yellow)                            [number] [default: 60]
  --style, -s             Badge style   [string] [choices: "plastic", "flat-square", "flat", "social"] [default: "flat"]
  --text, -t              Text on the left side of badge                                  [string] [default: "coverage"]
```


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [0.3.1](https://github.com/Morishiri/simple-coverage-badger/compare/v0.3.0...v0.3.1) (2020-07-20)

## [0.3.0](https://github.com/Morishiri/simple-coverage-badger/compare/v0.2.6...v0.3.0) (2020-07-20)


### ⚠ BREAKING CHANGES

* **ci:** Drop support for Node 8

### feature

* **ci:** Disable travis CI for node 8 ([5f8c7a0](https://github.com/Morishiri/simple-coverage-badger/commit/5f8c7a04500e692727eea33bc26bf28350f4dacb))

### [0.2.6](https://github.com/Morishiri/simple-coverage-badger/compare/v0.2.4...v0.2.6) (2020-07-20)

<a name="0.2.4"></a>
## [0.2.4](https://github.com/Morishiri/simple-coverage-badger/compare/v0.2.3...v0.2.4) (2017-09-30)


### Bug Fixes

* **cli:** default callback added ([7dc9b3e](https://github.com/Morishiri/simple-coverage-badger/commit/7dc9b3e))



<a name="0.2.3"></a>
## [0.2.3](https://github.com/Morishiri/simple-coverage-badger/compare/v0.2.2...v0.2.3) (2017-09-30)


### Bug Fixes

* **package:** fixed package.json entries to support global package ([52832dd](https://github.com/Morishiri/simple-coverage-badger/commit/52832dd))



<a name="0.2.2"></a>
## [0.2.2](https://github.com/Morishiri/simple-coverage-badger/compare/v0.2.1...v0.2.2) (2017-09-30)


### Bug Fixes

* **polyfill:** Object.values require ([832bb2e](https://github.com/Morishiri/simple-coverage-badger/commit/832bb2e))



<a name="0.2.1"></a>
## [0.2.1](https://github.com/Morishiri/simple-coverage-badger/compare/v0.2.0...v0.2.1) (2017-09-30)


### Bug Fixes

* **polyfill:** added Object.values polyfill for node 6 ([514a555](https://github.com/Morishiri/simple-coverage-badger/commit/514a555))



<a name="0.2.0"></a>
# 0.2.0 (2017-09-30)


### Bug Fixes

* **ci:** added travis config ([cc11118](https://github.com/Morishiri/simple-coverage-badger/commit/cc11118))
* **tests:** tests added, implementation adjusted ([d8e210c](https://github.com/Morishiri/simple-coverage-badger/commit/d8e210c))


### Features

* initialized repository (basic functionalities) ([dbf7327](https://github.com/Morishiri/simple-coverage-badger/commit/dbf7327))


lib/args.js
const yargs = require('yargs');
const values = require('object.values');

if (!Object.values) {
    values.shim();
}

const formats = require('./formats');
const scopes = require('./scopes');
const styles = require('./styles');


const defaultHosting = 'https://img.shields.io';

module.exports = yargs
    .option('input', {
        describe: 'Input file path',
        default: './coverage/cobertura-coverage.xml',
        alias: 'i',
        string: true
    })
    .option('outputFileName', {
        describe: 'Output file name (without extension)',
        default: 'badge',
        alias: 'o',
        string: true
    })
    .option('outputDirectory', {
        describe: 'Output directory',
        default: '.',
        alias: 'd',
        string: true
    })
    .option('format', {
        describe: 'Output file format (extension)',
        default: formats.SVG,
        alias: 'f',
        string: true,
        choices: Object.values(formats)
    })
    .option('badgeHosting', {
        describe: 'Badge hosting server',
        default: defaultHosting,
        alias: 'h',
        string: true
    })
    .option('coverageScope', {
        describe: 'Scope for badge',
        default: scopes.LINES,
        alias: 'c',
        string: true,
        choices: Object.values(scopes)
    })
    .option('excelentTreshold', {
        describe: 'Treshold for excelent coverage rating (brightgreen)',
        default: 95,
        alias: 'e',
        number: true
    })
    .option('goodTreshold', {
        describe: 'Treshold for good coverage rating (green)',
        default: 80,
        alias: 'g',
        number: true
    })
    .option('mediumTreshold', {
        describe: 'Treshold for medium coverage rating (yellow)',
        default: 60,
        alias: 'm',
        number: true
    })
    .option('style', {
        describe: 'Badge style',
        default: styles.FLAT,
        alias: 's',
        string: true,
        choices: Object.values(styles)
    })
    .option('text', {
        describe: 'Text on the left side of badge',
        default: 'coverage',
        alias: 't',
        string: true
    })
    .check((argv) => {
        if (argv.excelentTreshold < argv.goodTreshold)
            throw new Error('Excelent treshold cannot be smaller than good treshold');
        if (argv.excelentTreshold < argv.mediumTreshold)
            throw new Error('Excelent treshold cannot be smaller than medium treshold');
        if (argv.goodTreshold < argv.mediumTreshold)
            throw new Error('Good treshold cannot be smaller than medium treshold');
        return true;
    })
    .wrap(120)
    .argv;
